docs/release-history.rst

```diff
@@ -0,0 +1,13 @@
+***************
+Release history
+***************
+
+.. This extra heading is here to add an extra layer of depth to the TOC, so that
+   all the date headings don't end up filling up the sidebar toc tree
+   TODO: there must be a cleaner way to do this
```

Comment:
I'd probably cross off 1, because it is kind of nice to have those as actual headings in other TOCs (e.g. on `project.html`). 

Re: 2, I think that'd only let us control the depth of the `project.html` page as a whole (since that's where the TOC directive containing the changelog is). But the problem is that we want to exclude level 2 headings from the TOC in `release-history.html`, but not in the other pages linked from the `project.html` TOC.

For 3, I'm not sure whether that'd inevitably run into the same problem as 2 or not. I.e., perhaps it's impossible in sphinx to have different elements in a TOC have different depths. Or maybe that is a thing you can make work with a theme 🤷‍♂.

Reply:
For 2., the thread that pointed me to fulltoc was https://github.com/sphinx-doc/sphinx/issues/6033 and that seems to suggest that you can set a specific tocdepth for the sidebar on individual pages. But I could be misinterpreting htat.

For 3., I'm pretty confident that it's doable. What you would be doing is writing a directive that you would include on the `release-history.rst` page that modifies the depth of the sidebar's TOC. It is already possible to configure this globally throughout the theme (see https://github.com/nengo/nengo-sphinx-theme/blob/master/nengo_sphinx_theme/theme/sidebar.html#L52), but not on a per-page basis.

But yeah, I'm thinking that the eventual cost of evaluating and incorporating `fulltoc` is similar to or more than the cost of just doing it in Nengo Sphinx Theme, so I'll make an issue for that and we'll merge this in as is for now.